### DISCUSSION TITLE
Change keepUnusedDataFor after initialization

### DISCUSSION THREAD

**Discussion by ZimnitskiYury** [reduxjs/redux-toolkit#2532] [Q&A] [answered]
I want to grab this value from db. How i can change keepUnusedDataFor in rtk-query config after initialization?

**Comment by phryneas** [comment, marked as answer]
At the moment you can't . That would also require some quite complex logic internally. Can you explain your use case?

**Reply by ZimnitskiYury** [reply]
The customer wants to have control over cache lifetime in app settings :(

**Reply by phryneas** [reply]
All I could imagine is setting some kind of localStorage value and reloading the page. I'm sorry, but so far that sounds like a pretty nonsensical requirement and it's probably not a good idea to ship lots of extra code to the library consumers for that

**Reply by ZimnitskiYury** [reply]
> All I could imagine is setting some kind of localStorage value and reloading the page. I'm sorry, but so far that sounds like a pretty nonsensical requirement and it's probably not a good idea to ship lots of extra code to the library consumers for that

thx for the answer and advice!)
